### PR TITLE
Add tests for API method tcms.bugs.api.add_tag(). Refs #1597

### PR DESCRIPTION
**Pull request by mfonism**
This PR adds tests for the `add_tag()` api method in `tcms/bugs/api.py` as per #1597

### CODE REVIEW

File: tcms/bugs/models.py [hunk 1]
```diff
@@ -32,6 +32,11 @@ class Bug(models.Model, UrlMixin):
     executions = models.ManyToManyField('testruns.TestExecution', related_name='bugs')
     tags = models.ManyToManyField('management.Tag', related_name='bugs')
 
+    class Meta:
+        permissions = (
+            ('add_bugtag', 'can add bug tag'),
+        )
+
```
Comment (atodorov) [inline tcms/bugs/models.py:39]: There are 2 things going on in this PR. Should be 2 separate PRs or at least 2 separate commits.

There's a possible bug in how the tag permission is specified and handled and changes will affect more methods.

File: tcms/rpc/tests/utils.py [hunk 2]
```diff
@@ -29,3 +30,57 @@ def _fixture_setup(self):
...
+    def _fixture_setup(self):
+        # remove permission specified by `permission_label` from
+        # groups the api user is a member of
+        super()._fixture_setup()
+        self.remove_perm_from_api_user_groups(self.permission_label)
+
+    def remove_perm_from_api_user_groups(self, permission_label):
+        perm = self.get_perm_for_label(permission_label)
+        for group in self.api_user.groups.filter(permissions=perm):
+            group.permissions.remove(perm)
```
Comment (atodorov) [inline tcms/rpc/tests/utils.py:65]: Use `remove_perm_from_user()` don't duplicate it.

**Comment by mfonism** [comment]
@atodorov I've implemented the changes you requested

File: tcms/bugs/tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+class TestAddTag(APIPermissionsTestCase):
+    """Test Bug.add_tag"""
+
+    permission_label = 'bugs.add_bug_tags'
+
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.bug = BugFactory()
+        self.tag = TagFactory(name='rhel')
```
Comment (atodorov) [inline tcms/bugs/tests/test_api.py:23]: Don't hard-code the tag name

File: tcms/bugs/tests/test_api.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    def verify_api_with_permission(self):
+        self.rpc_client.Bug.add_tag(self.bug.pk, self.tag.name)
+        self.assertIn(self.tag, Bug.objects.get(pk=self.bug.pk).tags.all())
```
Comment (atodorov) [inline tcms/bugs/tests/test_api.py:27]: isn't `self.bug.tags.all()` the same ?

File: tcms/bugs/tests/test_api.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    def verify_api_without_permission(self):
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            self.rpc_client.Bug.add_tag(self.bug.pk, self.tag.name)
+
```
Comment (atodorov) [inline tcms/bugs/tests/test_api.py:32]: The scenarios below are functional testing, not related to permissions. They should be in a separate test class.

File: tcms/bugs/tests/test_api.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    def test_add_non_existent_tag_with_add_tag_perm(self):
+        user_should_have_perm(self.tester, self.permission_label)
+        user_should_have_perm(self.tester, 'management.add_tag')
+
+        self.assertFalse(Tag.objects.filter(name='fedora').exists())
+        self.rpc_client.Bug.add_tag(self.bug.pk, 'fedora')
+        self.assertTrue(Tag.objects.filter(name='fedora').exists())
+        self.assertTrue(self.bug.tags.filter(name='fedora').exists())
+
+    def test_add_non_existent_tag_without_add_tags_perm(self):
```
Comment (atodorov) [inline tcms/bugs/tests/test_api.py:47]: Validation of permission handling for `Tag.get_or_create` should be done separately as a test scenario for the Tag model. The API assumes that method works and does what it is supposed to do.

FTR this permission only controls auto-creation of tags.

**Comment by mfonism** [comment]
@atodorov I've implemented the changes you requested

File: tcms/bugs/tests/test_api.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
+# pylint: disable=attribute-defined-outside-init
+# pylint: disable=wrong-import-position
```
Comment (atodorov) [inline tcms/bugs/tests/test_api.py:2]: wrong-import-order, see pylint log in CI
